app/javascript/packs/foremCreatorSignup.js

```diff
@@ -0,0 +1,98 @@
+function setDefaultUsername(event) {
+  if (
+    document
+      .getElementsByClassName('js-creator-signup-username-row')[0]
+      .classList.contains('hidden')
+  ) {
+    const name = event.target.value;
+    // It's the first user and so we can assume that this username is not taken.
+    const usernameHint = createUsernameHint(name);
+    setUsernameHint(usernameHint);
+    setUsernameField(usernameHint);
+    showHintRow();
+  }
+}
+
+function createUsernameHint(name) {
+  return name
+    .toLowerCase()
+    .replace(/[^a-zA-Z0-9]/g, '_')
+    .substr(0, 30);
+}
+
+function showHintRow() {
+  const hintRow = document.getElementsByClassName(
+    'js-creator-signup-username-hint-row',
+  )[0];
```

Comment:
Refactor (blocking): For all `document.getElementsByClassName`

```javascript
document.getElementsByClassName(
    'some-selector',
  )[0];
```

opt for 

```javascript
document.querySelector(
    'some-selector',
  );
```

`getElementsByClassName` is technically faster it's more verbose and I would think over time browsers will optimize this.

TLDR it's an unnecessary optimization.

Reply:
@nickytonline there was [this PR](https://github.com/forem/forem/pull/11409) and some other comments over the past year where we made a concerted effort to use `getElementsByClassName` and it feels like moving these back to `querySelector` is reverting those efforts especially with the performance advantage. 

As per that PR I assumed the norm was:
>The querySelector* group should be favored only when actual CSS selectors (other than ID, class name and tag name) are needed.

Perhaps this should be a bigger discussion and we should establish some norms around it, instead of doing it on a per case basis.

Reply:
It can be a bigger discussion, but I won't die on this hill, so I'm fine with this.

Reply:
> Refactor (blocking):
> TLDR it's an unnecessary optimization.

IMO "blocking" and "unnecessary" do not go well together.